pkg/landscaper/installations/helper.go

```diff
@@ -174,6 +174,45 @@ func GetTargetImport(ctx context.Context, op lsoperation.Interface, contextName
 	return target, owner, nil
 }
 
+// GetTargetListImportByNames fetches the target imports from the cluster, based on a list of target names.
+func GetTargetListImportByNames(ctx context.Context, op lsoperation.Interface, contextName string, inst *Installation, targetNames []string) (*dataobjects.TargetList, []*v1.OwnerReference, error) {
+	targets := make([]lsv1alpha1.Target, len(targetNames))
+	owners := make([]*v1.OwnerReference, len(targetNames))
+	for i, targetName := range targetNames {
+		// get deploy item from current context
+		raw := &lsv1alpha1.Target{}
+		targetName = helper.GenerateDataObjectName(contextName, targetName)
+		if err := op.Client().Get(ctx, kubernetes.ObjectKey(targetName, inst.Info.Namespace), raw); err != nil {
+			return nil, nil, err
+		}
+		targets[i] = *raw
+		owners[i] = kubernetes.GetOwner(raw.ObjectMeta)
```

Comment:
maybe it makes sense to have the owner as part of the dataobject?
WDYT?

Reply:
I like this idea, will have a look at it.